src/mission_companion.cpp

```diff
@@ -1794,6 +1794,10 @@ npc_ptr talk_function::companion_choose( const std::string &skill_tested, int sk
         npc_companion_mission c_mission = guy->get_companion_mission();
         if( g->u.sees( guy->pos() ) && guy->is_friend() && c_mission.role_id.empty() ) {
             available.push_back( guy );
+        } else if( g->u.sees( guy->pos() ) && guy->mission == NPC_MISSION_GUARD_ALLY &&
+                   guy->companion_mission_role_id != "FACTION_CAMP" && g->u.posz() == guy->posz() &&
+                   rl_dist( g->u.pos(), guy->pos() ) <= SEEX * 2 ) {
```

Comment:
Do you need both the position check and g->u.sees( guy->pos() )?
Also, consider:
```
if( g->u.sees( guy->pos() ) ) {
    if( guy->is_friend() && c_mission.role_id.empty() ) {
        available.push_back( guy );
    } else if( guy->mission == NPC_MISSION_GUARD_ALLY &&
                   guy->companion_mission_role_id != "FACTION_CAMP" ) {
        available.push_back( guy );
    }
```
or really just combine it all into one statement but do whatever seems most likely to be understood when you review this code in 4 months.

Reply:
I thought the sees check would work on it's own, but it didnt, it was throwing up the option for an off-screen NPC to be chosen. I dont know why.
Obviously im doing something wrong, and I shouldnt need both checks so ill try and wrangle it some more tomorrow.

Reply:
Yeah that structure fixed it.